cli/cmd/org/upload_logo.go

```diff
@@ -0,0 +1,150 @@
+package org
+
+import (
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/rilldata/rill/cli/pkg/cmdutil"
+	adminv1 "github.com/rilldata/rill/proto/gen/rill/admin/v1"
+	"github.com/spf13/cobra"
+)
+
+const maxImageSize = 3 * 1024 * 1024 // 3 MB
+
+func UploadLogoCmd(ch *cmdutil.Helper) *cobra.Command {
+	var path string
+	var remove bool
+
+	cmd := &cobra.Command{
+		Use:   "upload-logo [<org-name> [<path-to-image>]]",
+		Args:  cobra.MaximumNArgs(2),
+		Short: "Upload a custom logo",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			client, err := ch.Client()
+			if err != nil {
+				return err
+			}
+
+			// Parse positional args into flags
+			if len(args) > 0 {
+				ch.Org = args[0]
+				if len(args) > 1 {
+					path = args[1]
+				}
+			}
+			if ch.Org == "" {
+				return fmt.Errorf("an organization name is required")
+			}
+
+			// Handle --remove
+			if remove {
+				if path != "" {
+					return fmt.Errorf("cannot specify both --remove and a path")
+				}
+
+				// Confirmation prompt
+				if ok, err := cmdutil.ConfirmPrompt(fmt.Sprintf("You are removing the custom logo for %q. Continue?", ch.Org), "", false); err != nil || !ok {
+					return err
+				}
+
+				empty := ""
+				_, err = client.UpdateOrganization(cmd.Context(), &adminv1.UpdateOrganizationRequest{
+					Name:        ch.Org,
+					LogoAssetId: &empty,
+				})
+				if err != nil {
+					return err
+				}
+
+				ch.PrintfSuccess("Removed logo from organization %q\n", ch.Org)
+				return nil
+			}
+
+			// Check the file is an image
+			ext := strings.TrimPrefix(filepath.Ext(path), ".")
+			switch ext {
+			case "png", "jpg", "jpeg":
+			default:
+				return fmt.Errorf("invalid file type %q (expected PNG or JPG)", ext)
+			}
+
+			// Validate and open the path
+			fi, err := os.Stat(path)
+			if err != nil {
+				return fmt.Errorf("failed to read %q: %w", path, err)
+			}
+			if fi.IsDir() {
+				return fmt.Errorf("failed to upload %q: the path is a directory", path)
+			}
+			if fi.Size() == 0 {
+				return fmt.Errorf("failed to upload %q: the file is empty", path)
+			}
+			if fi.Size() > maxImageSize {
+				return fmt.Errorf("failed to upload %q: the file size exceeds the limit of %d bytes", path, maxImageSize)
+			}
+			f, err := os.Open(path)
+			if err != nil {
+				return fmt.Errorf("failed to open %q: %w", path, err)
+			}
+			defer f.Close()
+
+			// Confirmation prompt
+			if ok, err := cmdutil.ConfirmPrompt(fmt.Sprintf("You are changing the custom logo for %q. Continue?", ch.Org), "", false); err != nil || !ok {
+				return err
+			}
+
+			// Generate the asset upload URL
+			asset, err := client.CreateAsset(cmd.Context(), &adminv1.CreateAssetRequest{
```

Comment:
can you not follow the network calls to upload a logo of size less than 100 MB (signed url header limit)

Reply:
I just intended it as a convenience to avoid uploading an image that would load very slowly, but I agree it's probably safer and more future proof to enforce the size on the backend. I pushed a change that enforces different upload limits on the backend depending on whether it's a deploy archive or an image.